sound/soc/sof/intel/hda.c

```diff
@@ -407,16 +407,26 @@ static irqreturn_t hda_dsp_interrupt_handler(int irq, void *context)
 	struct snd_sof_dev *sdev = context;
 	struct sof_intel_hda_dev *hdev = sdev->pdata->hw_pdata;
 
-	if (hda_dsp_check_ipc_irq(sdev) ||
-	    hda_dsp_check_stream_irq(sdev))
-		return IRQ_WAKE_THREAD;
+	/*
+	 * hda_dsp_check_ipc_irq will disable IPC interrupt if
+	 * ipc irq is triggered
+	 */
+	if (hda_dsp_check_ipc_irq(sdev))
+		sdev->irq_event |= SOF_IRQ_IPC;
+
+	//TODO: disable stream interrupt
```

Comment:
@bardliao I'm not sure we should disable the stream interrupt. we also use the same INTSTS to check for codec unsolicited responses and we loop 10 times over to do this. If we disable the stream interrupt, that might not be needed anymore and I feel like this could end up causing IPC timeout regressions.